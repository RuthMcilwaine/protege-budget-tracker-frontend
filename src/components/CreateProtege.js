import React, { Component } from 'react';
import { Redirect } from 'react-router-dom'
// import axios from 'axios'


class CreateProtege extends Component {
  state = { 
    protegename: '',
    protegeemail: '',
    expenditure: '',
    balance: 1000,
    addingUser: true
  }

  handleClick = (e) => {
    e.preventDefault()
    const protegeData = {
      protegename: this.state.protegename,
      protegeemail: this.state.protegeemail,
      expenditure: this.state.expenditure,
      balance: this.state.balance
    }
    this.props.addProtege(protegeData)
  }

  handleCancel = (e) => {
    e.preventDefault();
    window.location = '/proteges';
  }

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value})
  }

  render(){
    // console.log(this.props.addingUser)
    if (!this.props.addingUser) {
      return <Redirect to='/' />
    } else {
      return (
        <div className="protege-form">
          <h1>Protege form</h1>
          <form>
            <label htmlFor="protegename">Protege name</label>
              <input 
                onChange={this.handleChange} 
                type="text" 
                id="protegename" 
                placeholder="protege name" 
                value={this.state.protegename} 
              />
              <label htmlFor="protegeemail">Email</label>
              <input 
                onChange={this.handleChange} 
                type="text" 
                id="protegeemail" 
                placeholder="email" 
                value={this.state.protegeemail} 
              />
              <button onClick={this.handleClick}>Save</button>
              <button onClick={this.handleCancel}>Cancel</button>
          </form>
        </div>
      )
    }
  }
}


  export default CreateProtege;