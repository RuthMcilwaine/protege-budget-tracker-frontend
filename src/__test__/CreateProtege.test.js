import React from 'react';
import ReactDOM from 'react-dom';
import CreateProtege from '../components/CreateProtege';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

const protegeData = [{
  "protege": {
  "_id": "5d26b687a7946c6d37dcfdde",
  "protegename": "El",
  "protegeemail": "Elizabeth@myob.com.au",
  "expenditure": 0,
  "balance": 1000,
  "createdAt": "2019-07-11T04:09:43.804Z",
  "updatedAt": "2019-07-18T00:37:55.952Z",
  "__v": 0
  }}
]

describe('<CreateProtege />', () => {
  it('renders <ProtegeData /> components', () => {
    const wrapper = shallow(<CreateProtege protegeData={protegeData}/>);
  });
  expect(protegeData).toMatchSnapshot();
})

