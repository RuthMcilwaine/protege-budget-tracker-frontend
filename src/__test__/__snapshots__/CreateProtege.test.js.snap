// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  Object {
    "protege": Object {
      "__v": 0,
      "_id": "5d26b687a7946c6d37dcfdde",
      "balance": 1000,
      "createdAt": "2019-07-11T04:09:43.804Z",
      "expenditure": 0,
      "protegeemail": "Elizabeth@myob.com.au",
      "protegename": "El",
      "updatedAt": "2019-07-18T00:37:55.952Z",
    },
  },
]
`;
